src/ensembl/src/content/app/species/state/general/speciesGeneralHelper.ts

```diff
@@ -0,0 +1,419 @@
+import { buildFocusIdForUrl } from 'src/shared/state/ens-object/ensObjectHelpers';
+import { urlObj } from 'src/shared/components/view-in-app/ViewInApp';
+
+export enum SpeciesStatsSections {
+  CODING_STATS = 'coding_stats',
+  NON_CODING_STATS = 'non_coding_stats',
+  PSEUDOGENES = 'pseudogene_stats',
+  ASSEMBLY_STATS = 'assembly_stats'
+}
+
+// SpeciesStatsSections -> Groups
+enum Groups {
+  CODING_GENES = 'Coding genes',
+  NON_CODING_GENES = 'Non-coding genes',
+  ANALYSIS = 'Analysis',
+  PSEUDOGENES = 'Pseudogenes',
+  ASSEMBLY_STATS = 'Assembly'
+}
+
+// SpeciesStatsSections -> Groups -> Stats
+enum Stats {
+  // Coding stats
+  CODING_GENES = 'Coding genes',
+  SHORTEST_GENE = 'Shortest gene',
+  LONGEST_GENE = 'Longest gene',
+  AVERAGE_GENOMIC_SPAN = 'Average genomic span',
+  AVERAGE_SEQUENCE_LENGTH = 'Average sequence length',
+  AVERAGE_CDS_LENGTH = 'Average CDS length',
+  TOTAL_TRANSCRIPTS = 'Total transcripts',
+  CODING_TRANSCRIPTS = 'Coding transcripts',
+  TRANSCRIPTS_PER_GENE = 'Transcripts per gene',
+  CODING_TRANSCRIPTS_PER_GENE = 'Coding transcripts per gene',
+  TOTAL_EXONS = 'Total exons',
+  TOTAL_CODING_EXONS = 'Total coding exons',
+  AVERAGE_EXON_LENGTH = 'Average exon length',
+  AVERAGE_CODING_EXON_LENGTH = 'Average coding exon length',
+  AVERAGE_EXONS_PER_TRANSCRIPT = 'Average exons per transcript',
+  AVERAGE_CODING_EXONS_PER_CODING_TRANSCRIPT = 'Average coding exons per coding transcript',
+  TOTAL_INTRONS = 'Total introns',
+  AVERAGE_INTRON_LENGTH = 'Average intron length',
+
+  // Assembly Stats
+  CHROMOSOMES = 'Chromosomes',
+  TOTAL_GENOME_LENGTH = 'Total genome length',
+  TOTAL_CODING_SEQUENCE_LENGTH = 'Total coding sequence length',
+  TOTAL_GAP_LENGTH = 'Total gap length',
+  SPANNED_GAP = 'Spanned gaps',
+  TOPLEVEL_SEQUENCES = 'Toplevel sequences',
+  COMPONENT_SEQUENCES = 'Component sequences',
+  AVERAGE_GC_CONTENT = '% GC',
+  ASSEMBLY_NAME = 'Assembly name',
+  ASSEMBLY_DATE = 'Assembly date',
+  ASSEMBLY_ACCESSION = 'Assembly accession',
+  CONTIG_N50 = 'Contig N50',
+  TAXONOMY_ID = 'Taxonomy id',
+  BREED_CULTIVAR_STRAIN = 'Breed/Cultivar/Strain',
+  SEX = 'Sex',
+  SCIENTIFIC_NAME = 'Scientific name',
+
+  // Non coding stats
+  NON_CODING_GENES = 'Non-coding genes',
+  SMALL_NON_CODING_GENES = 'Small non-coding genes',
+  LONG_NON_CODING_GENES = 'Long non-coding genes',
+  MISC_NON_CODING_GENES = 'Misc non-coding genes',
+
+  // Psuedogene stats
+  PSEUDOGENES = 'Pseudogenes'
+}
+
+type SpeciesStatsSectionGroups = {
+  [key in SpeciesStatsSections]: {
+    title: Stats | string;
+    groups: Groups[];
+    primaryStatsKey?: Stats;
+    secondaryStatsKey?: Stats;
+    hasExampleLink?: boolean;
+    exampleLinkText?: string;
+  };
+};
+
+// Maps the groups to it's respective section
+export const sectionGroupsMap: SpeciesStatsSectionGroups = {
+  [SpeciesStatsSections.CODING_STATS]: {
+    title: Stats.CODING_GENES,
+    hasExampleLink: true,
+    exampleLinkText: 'Example gene',
```

Comment:
`hasExampleLink` and `exampleLinkText` seem always to coincide. `hasExampleLink` looks like a redundant property; let's get rid of it

Reply:
Yep.

Reply:
That's done